pkg/deployment/depresource/new_deployment.go

```diff
@@ -0,0 +1,135 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package depresource
+
+import (
+	"io"
+
+	"github.com/elastic/cloud-sdk-go/pkg/api"
+	"github.com/elastic/cloud-sdk-go/pkg/models"
+)
+
+// NewParams is consumed by New()
+type NewParams struct {
+	*api.API
+
+	Name                   string
+	Version                string
+	DeploymentTemplateID   string
+	Region                 string
+	ElasticsearchRefID     string
+	KibanaRefID            string
+	ApmRefID               string
+	AppsearchRefID         string
+	ElasticsearchSize      int32
+	ElasticsearchZoneCount int32
+	KibanaSize             int32
+	KibanaZoneCount        int32
+	ApmSize                int32
```

Comment:
I wonder if we can group the same per cluster type attributes in a structure to increase readability
I can see there are attributes like `size`, `zoneCount` and maybe `enable` that are repeated